_problems/fee-bumping.md

```diff
@@ -0,0 +1,96 @@
+
+For layer-2 protocols this rules had the unintended consequence of introducing  "pinning" attacks (see [Related Research](#related-research)).
+This is usually done by adding a lot of descendants to a transaction that conflicts with the one the honest party needs confirm.
+As we have said an honest party needs to confirm a transactions in a certain time -- if it can't evict a maliciously crafted transaction then they are in a helpless position.
+
+An instinctive exception to [BIP125]'s rules called the [CPFP carve out] was introduced to allow lightning designers to avoid pinning attacks with some extra design effort.
+
+## Impact
+
+The principles layer-2 protocol designers should use to enable transactions to confirm on time are an esoteric domain understood by only a handful of people.
+Even those practiced in the occult [have made mistakes](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-September/002796.html) trying to design around them.
+Having a simple set of principles would increase the confidence in the resulting protocols and allow a wider group to understand and build layer-2 systems.
+
+## Potential Directions
+
+1. **Reconsidering mempool and relay rules**: It is possible that mempool rules could be made less problematic for layer-2 protocol designers without harming the Bitcoin system.
+   The main concern with this approach is that the rules that are optimal for layer-2 protocols may not be the rules that are optimal for miner profit or preventing DoS attacks.
+   However, if tweaked relay rules were to allow layer-2 Bitcoin protocols to thrive then the increased value of the Bitcoin network could more than compensate miners for using a slightly suboptimal algorithm (as long as they don't introduce DoS attacks).
+2. **Consensus changes**: It may be possible introduce new ways of increasing an already broadcasted transaction's fee through a consensus change (see [Proposed Solutions](#proposed-solutions)).
+
+## Proposed Solutions
+
+In *[A Replacement for RBF and CPFP: Non-Destructive TXID Dependencies for Fee Sponsoring]* Rubin proposes allowing a transaction to sponsor another.
+The sponsor transaction can only be included in a block if the transaction it is sponsoring is in it too.
+The sponsor transaction's fees then increase th overall feerate of the pair of transactions.
+The advantage of this idea is that it is always available to any transaction without the protocol designer having to consider fee bumping as part of the design.
+
+## Related Research
+
+1. In *[RBF Pinning with Counterparties and Competing Interest]* Corallo introduces the idea of a pinning attack.
+2. In *[Pinning : The Good, The Bad, The Ugly]* Riard describes the problem of pinning attacks and gives several examples relevant to Lightning.
+3. The *[Mempool and mining]* article on the Bitcoin core development wiki Daftuar provides the rationale for why the mempool rules are the way they are.
```

Comment:
The OG pinning mitigated by the carve-out mechanism : https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-November/016518.html